src/hexdoc/patchouli/page/pages.py

```diff
@@ -78,6 +79,19 @@ class Multiblock(HexdocModel):
     symmetrical: bool = False
     offset: tuple[int, int, int] | None = None
 
+    @property
+    def bill_of_materials(self) -> list[tuple[ItemWithTexture | TagWithTexture, int]]:
+        bom = defaultdict[str, int](int)
+        for layer in self.pattern:
+            for row in layer:
+                for item in row:
+                    if item in self.mapping:
```

Comment:
the only cases where this would be false are ` ` (air), `_` (any block), or an invalid page. imo we should ignore air, handle "any block" somehow, and raise otherwise if it's not found

Reply:
also `0` to mark the center or whatever it is ? I didn't know about the `_` any block though.

Reply:
aha true! ...actually i can probably deal with this one if you want, many edge cases apparently